### PR TITLE
feat: Migrate to Keycloak ADR

### PR DESCRIPTION
**Pull request by tumido**
Reverts operate-first/blueprint#102 which is a revert of operate-first/blueprint#101

Let's review it properly now :smile:

### CODE REVIEW

File: docs/adr/0018-migration-to-keycloak.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+1. Authenticate OpenShift against GitHub directly and provide app authentication via Keycloak
+2. Authenticate OpenShift against GitHub directly and provide app authentication via Dex against OpenShift
+3. Authenticate OpenShift and apps against Keycloak
+4. Authenticate OpenShift and apps against Keycloak and migrate OpenShift groups to Keycloak Groups
+5. Authenticate OpenShift against Keycloak and provide app authentication via Dex against OpenShift
+
+## Decision Outcome
+
+Chosen option: _"4. Authenticate OpenShift against Keycloak and provide app authentication via Dex against OpenShift"_, because it's a fully declarative solution that is fully supported and available today.
```
Comment (HumairAK) [inline docs/adr/0018-migration-to-keycloak.md]: Typo -- chosen option is actually 5 it seems :)

File: docs/adr/0018-migration-to-keycloak.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+This ADR should be revisited after Keycloak became capable of importing/providing OpenShift Groups data ([Keycloak RFE 18532][1]). Then option _3. Authenticate OpenShift and apps against Keycloak_ should be preferred.
```
Comment (HumairAK) [inline docs/adr/0018-migration-to-keycloak.md]: became > becomes

**Comment by HumairAK** [comment]
besides above comments:
/lgtm

**Review by 4n4nd — APPROVED** [review]
/lgtm
Looks great!

**Pull request merged**